sound/soc/sof/sof-audio.c

```diff
@@ -103,7 +103,7 @@ int sof_widget_free(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget)
 		.id = swidget->comp_id,
 	};
 	struct sof_ipc_reply reply;
-	int ret, ret1;
+	int ret, err;
```

Comment:
commit message: it's really unclear what this patch fixes.

You already had a comment before
````c
                 * The scheduler widget for a pipeline is not part of the connected DAPM
		 * widget list and it needs to be set up before the widgets in the pipeline
		 * are set up.
````

So what changes with this patch?

Does this need a Fixes: tag? is this a fixup! ?

Reply:
No it doesnt fix anything for the SOF firmware with IPC3. But for IPC4, this is a requirement. The pipeline has to be set up before the widgets are. And because we set the BE DAI widget in the BE hw_params and BE hw_params happens before FW hw_params (which is where we set up the pipe widget today), we need to change this.

Reply:
Not getting the idea. why did we have this comment in the public tree which only supports IPCv3 as of today?

the comment you added dates from 5fcdbb2d45df ('ASoC: SOF: Add support for dynamic pipelines'), so if you change the code was that commit not broken?

Reply:
it was added for the purpose of supporting IPC4 @plbossart but this was a miss.

Reply:
> it was added for the purpose of supporting IPC4 @plbossart but this was a miss.

so if it's a miss, why not add the Fixes tag then, and explain what was missed?

Reply:
ok done